### DISCUSSION TITLE
What is the true spec of the response for `getAlbumList`

### DISCUSSION THREAD

**Discussion by gabeklavans** [opensubsonic/open-subsonic-api#90] [General]
Specifically, `getAlbumList`, not `getAlbumList2`.

In the [example response](https://opensubsonic.netlify.app/docs/responses/albumlist/), `songCount` is included.

However, the spec says that `album` is an Array of [Child](https://opensubsonic.netlify.app/docs/responses/child), and `Child` does not have a `songCount` field defined, neither required nor optional.

So, should the example for `getAlbumList` be changed to remove `songCount`, or should `Child` be updated to include `songCount`? I'm leaning towards the second option, but I'm not sure exactly where `Child` fits into the API at this moment.

**Comment by gabeklavans** [comment]
Adding on a related question, for `getAlbumList2`, the [example response for `getAlbumList2`](https://opensubsonic.netlify.app/docs/endpoints/getalbumlist2/) and [`albumList2`](https://opensubsonic.netlify.app/docs/responses/albumlist2/) shows `isDir` and `parent` in each returned object, but the spec for [AlbumID3](https://opensubsonic.netlify.app/docs/responses/albumid3/) doesn't contain either of these. So I ask the same question for these fields: remove from the example or add to the item spec?

**Reply by Tolriq** [reply]
The examples are valid data so can't be modified. It's possible the return type are different and that some servers use different ones too internally.

https://www.subsonic.org/pages/api.jsp expose most of the original API.

**Reply by dweymouth** [reply]
I think @gabeklavans is right about `getAlbumList2` - the original spec says it returns a list of `AlbumID3` type which doesn't have `isDir` and `parent`. See the example response from the official API docs:

https://www.subsonic.org/pages/inc/api/examples/albumList2_example_1.xml

**Reply by Tolriq** [reply]
The official reference is https://www.subsonic.org/pages/inc/api/schema/subsonic-rest-api-1.16.1.xsd so whatever is there is the source of truth.

As I said in previous issue, it took me an insane amount of time to migrate the docs and no one help finished when I asked for help so if you make PR that match the xsd they are valid whatever the changes position are.

**Reply by gabeklavans** [reply]
> As I said in previous issue, it took me an insane amount of time to migrate the docs and no one help finished when I asked for help

To be clear, I'm more than willing to start opening PRs. I just made it a discussion because I wasn't aware of the source of truth.

> The official reference is https://www.subsonic.org/pages/inc/api/schema/subsonic-rest-api-1.16.1.xsd so whatever is there is the source of truth.

Thanks for this, I'll start digging into it

**Reply by gabeklavans** [reply]
> The examples are valid data so can't be modified. It's possible the return type are different and that some servers use different ones too internally.
> 
> 
> 
> https://www.subsonic.org/pages/api.jsp expose most of the original API.

It seems though that the `Child` object is an object added by OpenSubsonic (unless I'm misunderstanding the "[OS]" marker in the docs). Should it include `songCount` to match the example? @Tolriq

**Reply by dweymouth** [reply]
The [OS] marker means fields were added to the Child object in OpenSubsonic - it's not a new object. I don't think *songCount* has been added to child for OpenSubsonic, and it's a mistake that the example response includes it. Most of the new OpenSubsonic properties for albums are only added to AlbumID3, since that's what most modern clients are using.

**Comment by gabeklavans** [comment]
Resolved by #91
